content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md

```diff
@@ -662,53 +662,99 @@ updates:
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan="3" align="center"><code>^1.0.0</code></td>
+            <td rowspan="3" align="center"><code>1.0.0</code></td>
+            <td rowspan="3" align="center"><code>1.2.0</code></td>
+            <td align="center"><code>'widen'</code></td>
+            <td><code>^1.0.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase'</code></td>
+            <td><code>^1.2.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase-if-necessary'</code></td>
+            <td><code>^1.0.0</code></td>
+        </tr>
+      <tr>
+            <td rowspan="3" align="center"><code>^1.0.0</code></td>
+            <td rowspan="3" align="center"><code>1.0.0</code></td>
+            <td rowspan="3" align="center"><code>2.0.0</code></td>
+            <td align="center"><code>'widen'</code></td>
+            <td><code>&gt;=1.0.0 &lt;3.0.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase'</code></td>
+            <td><code>^2.0.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase-if-necessary'</code></td>
+            <td><code>^2.0.0</code></td>
+        </tr>
+    </tbody>
+</table>
 
 Use the `versioning-strategy` option to change this behavior for supported package managers.
 
 {% data reusables.dependabot.option-affects-security-updates %}
 
 Available update strategies
 
-| Option | Supported by | Action |
-|--------|--------------|--------|
-| `lockfile-only` | `bundler`, `cargo`, `composer`, `mix`, `npm`, `pip` | Only create pull requests to update lockfiles. Ignore any new versions that would require package manifest changes. |
-| `auto` | `bundler`, `cargo`, `composer`, `mix`, `npm`, `pip` | Follow the default strategy described above.|
-| `widen`| `composer`, `npm` | Relax the version requirement to include both the new and old version, when possible. |
-| `increase`| `bundler`, `composer`, `npm` | Always increase the version requirement to match the new version. |
-| `increase-if-necessary` | `bundler`, `composer`, `npm` | Increase the version requirement only when required by the new version. |
+| Ecosystem | Supported _versioning-strategies_ | Default strategy | Additional notes |
+|-----------|-----------------------------------|------------------|------------------|
+| `bundler` | `auto`, `increase`, `increase-if-necessary`, `lockfile-only` | `auto` |
+| `cargo` | `auto`, `lockfile-only` | `auto` |
+| `composer` | `auto`, `lockfile-only`, `widen`, | `auto` |
+| `docker` | TODO | TODO |
+| `github-actions` | TODO | TODO |
+| `gitsubmodule` | TODO | TODO |
+| `go` | TODO | TODO |
+| `gradle` | TODO | TODO |
+| `maven` | TODO | TODO |
+| `mix` | `auto`, `lockfile-only` | `auto` |
+| `npm` | `auto`, `lockfile-only`, `widen` | `auto` |
+| `nuget` | TODO | TODO |
+| `pip` | `auto`, `lockfile-only` | `auto` |
```

Comment:
```suggestion
| `pip` | `auto`, `increase`, `increase-if-necessary`, `lockfile-only`, `widen` | `auto` |
```
`increase` is definitely supported here, and with #6632 merged I think `widen` can be added as well?
`increase-if-necessary` I've never used myself, but from the error message I got when I tried to use `widen` I think it's supported as well:
`The property '#/updates/0/versioning-strategy' value "widen" did not match one of the following values: lockfile-only, auto, increase, increase-if-necessary`

Reply:
Yes, all those 4 are currently supported for Python projects, and I plan to work on adding `widen` soon.

I think you meant to link to https://github.com/dependabot/dependabot-core/pull/6632, which fixed the update logic to better support `widen`, but we still need some internal work to expose the option in the configuration file.

Reply:
I see, thanks for the correction. I updated the suggested change and the link in my comment.

Reply:
Yeah, I just tried on a test repo, and `widen` isn't yet supported, or at least the config file validator chokes on it. Thanks to @deivid-rodriguez 's hard work, some of the building blocks are in place, but there's a little more to do to get it across the finish line.